File: Backend/src/ai-waste-prediction-service/src/controllers/notificationController.js
const Notification = require("../models/Notification");

const getAllNotifications = async (req, res) => {
  try {
    const notifications = await Notification.find().sort({ createdAt: -1 });

    res.status(200).json({
      message: "Notifications fetched successfully",
      count: notifications.length,
      notifications,
    });
  } catch (error) {
    res.status(500).json({
      message: "Error fetching notifications",
      error: error.message,
    });
  }
};

const getNotificationsByBatchId = async (req, res) => {
  try {
    const { batchId } = req.params;

    const notifications = await Notification.find({ batchId }).sort({ createdAt: -1 });

    res.status(200).json({
      message: "Batch notifications fetched successfully",
      count: notifications.length,
      notifications,
    });
  } catch (error) {
    res.status(500).json({
      message: "Error fetching batch notifications",
      error: error.message,
    });
  }
};

module.exports = {
  getAllNotifications,
  getNotificationsByBatchId,
};

File: Backend/src/ai-waste-prediction-service/app/controllers/notification_controller.py
from app.config.db import notifications_collection


def serialize_notification(notification):
    notification["_id"] = str(notification["_id"])
    return notification


async def get_all_notifications():
    notifications = list(notifications_collection.find().sort("createdAt", -1))
    notifications = [serialize_notification(n) for n in notifications]

    return {
        "message": "Notifications fetched successfully",
        "count": len(notifications),
        "notifications": notifications,
    }


async def get_notifications_by_batch_id(batch_id: str):
    notifications = list(
        notifications_collection.find({"batchId": batch_id}).sort("createdAt", -1)
    )

    notifications = [serialize_notification(n) for n in notifications]

    return {
        "message": "Batch notifications fetched successfully",
        "count": len(notifications),
        "notifications": notifications,
    }
